src/cpp/lpnamer/main/ProblemGeneration.cpp

```diff
@@ -44,30 +43,57 @@ void CreateDirectories(const std::filesystem::path& output_path) {
 }
 
 ProblemGeneration::ProblemGeneration(ProblemGenerationOptions& options)
-    : options_(options) {}
+    : options_(options) {
+  if (!options_.StudyPath().empty()) {
+    mode_ = Mode::ANTARES_API;
+  } else if (!options_.XpansionOutputDir().empty()) {
+    mode_ = Mode::FILE;
+  } else if (!options_.ArchivePath().empty()) {
+    mode_ = Mode::ARCHIVE;
+  }
+}
+
 std::filesystem::path ProblemGeneration::updateProblems() {
-  const auto xpansion_output_dir = options_.XpansionOutputDir();
+  using namespace std::string_literals;
+  std::filesystem::path xpansion_output_dir;
   const auto archive_path = options_.ArchivePath();
 
-  auto deduced_xpansion_output_dir =
-      options_.deduceXpansionDirIfEmpty(xpansion_output_dir, archive_path);
+  if (mode_ == Mode::ARCHIVE) {
+    xpansion_output_dir =
+        options_.deduceXpansionDirIfEmpty(xpansion_output_dir, archive_path);
+  }
+
+  if (mode_ == Mode::ANTARES_API) {
+    throw std::runtime_error("Mode API Not implemented yet");
+  }
+
+  if (mode_ == Mode::FILE) {
+    simulation_dir_ = options_.XpansionOutputDir();  // Legacy naming.
+    // options_.XpansionOutputDir() point in fact to a simulation output from
+    // antares
+  }
+
+  if (mode_ == Mode::ANTARES_API || mode_ == Mode::FILE) {
+    xpansion_output_dir = simulation_dir_;
+  }
 
   const auto log_file_path =
-      deduced_xpansion_output_dir / "lp" / "ProblemGenerationLog.txt";
+      xpansion_output_dir / "lp"s / "ProblemGenerationLog.txt"s;
 
-  CreateDirectories(deduced_xpansion_output_dir);
+  CreateDirectories(xpansion_output_dir);  // Ca ou -Xpansion ?
```

Comment:
Does the Python part (or any other part of the code) rely on the `-Xpansion` suffix ?

Reply:
Yes, but the conditions are foggy